kotest-assertions/kotest-assertions-klock/src/commonMain/kotlin/io.kotest/matchers/date/date.kt

```diff
@@ -0,0 +1,136 @@
+package io.kotest.matchers.date
+
+import com.soywiz.klock.Date
+import com.soywiz.klock.DateTime
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+
+infix fun Date.shouldHaveSameYearAs(date: Date) = this shouldBe haveSameYear(date)
+
+infix fun Date.shouldNotHaveSameYearAs(date: Date) = this shouldNotBe haveSameYear(date)
+
+fun haveSameYear(date: Date) = object : Matcher<Date> {
+   override fun test(value: Date) =
+      MatcherResult(
+         value.year == date.year,
+         { "$value should have year ${date.year}" },
+         { "$value should not have year ${date.year}" }
+      )
+}
+
+infix fun Date.shouldHaveYear(year: Long) = this shouldBe haveYear(year)
```

Comment:
I think we should not take year as Long it should be Int, else while using we will have to explicitly need to convert Int to Long in tests like in example below
```kotlin
date shouldHaveSameYear 2020.toLong()
```
more preferable would be 
```kotlin
date shouldHaveSameYear 2020
```

Same would go for month, days, hours etc in this and Time matchers

Reply:
Yes, you are right. I will update the code.